knowledge_base/002_TNS_Could_Not_Resolve_Connect_Identifier_Error_Fix.md

```diff
@@ -0,0 +1,22 @@
+# "ORA-12154: TNS:could not resolve the connect identifier specified" Error
+
+## This error message commonly shows up for the following 2 reasons:
+1. Incorrect wallet configuration.
+   1. Follow [instructions here](https://github.com/oracle/oci-data-science-ai-samples/blob/master/environment_examples/database/autonomous_database.ipynb), if you haven't already, to download the wallet and unzip it locally.
+   2. Next, within `~/.database` directory there should be a folder with your database name. Within that will be the unzipped wallet files with some modifications made by `ads` code.  Here are two things to confirm are correct within the wallet files:
+      1. Check that `sqlnet.ora` file has the wallet location referenced. 
+      For example sqlnet.ora will have a part like `(METHOD_DATA = (DIRECTORY="/Users/rnahar/.database/Wallet_ADW")))`
+      2. Check the connection identifier is found in the `tns_names.ora`. 
+      For example there should be a section like `dbtest123_low = (description= (retry_count=20)(retry_delay=3)(address=(protocol=tcps)(port=1522)(host=...`
+      3. Lastly check connection with sql*plus. This will confirm username/password and wallet are valid. Template for command:
+      `sqlplus <user_name>/<password>@<protocol>://<host>:<port>/<service_name>?wallet_location="<unziped_wallet_location>"`
```

Comment:
You should define what each of these are and if possible give examples for things like the typical port number and that the unziped_wallet_location is the directory that was discussed above.

Reply:
Example added! And provided more information about the params